src/MongoDB.Driver/MongoCollectionImpl.cs

```diff
@@ -55,12 +55,7 @@ private MongoCollectionImpl(IMongoDatabase database, CollectionNamespace collect
             _operationExecutor = Ensure.IsNotNull(operationExecutor, nameof(operationExecutor));
             _documentSerializer = Ensure.IsNotNull(documentSerializer, nameof(documentSerializer));
 
-            _messageEncoderSettings = new MessageEncoderSettings
-            {
-                { MessageEncoderSettingsName.GuidRepresentation, _settings.GuidRepresentation },
-                { MessageEncoderSettingsName.ReadEncoding, _settings.ReadEncoding ?? Utf8Encodings.Strict },
-                { MessageEncoderSettingsName.WriteEncoding, _settings.WriteEncoding ?? Utf8Encodings.Strict }
-            };
+            _messageEncoderSettings = new Lazy<MessageEncoderSettings>(GetMessageEncoderSettings); // lazy needs to avoid infinite looping in encryption configuration
```

Comment:
Without the lazy object, we will create `LibMongoController` infinitely. Since we get keyvault `MongoCollection`  in the constructor. There are other solutions which I've tried like a new method `Initialize` in the controller, but probably this way is the simplest. We can discuss it.

Reply:
Agreed, that's it's not an ideal way